src/sentry/utils/jwt.py

```diff
@@ -0,0 +1,84 @@
+"""Common handling of JWT tokens.
+
+This is an attempt to have all the interactions with JWT in once place, so that we have once
+central place which handles JWT in a uniform way.
+"""
+
+from typing import List, Mapping, Optional, Union
+
+import jwt as pyjwt
+from cryptography.hazmat.primitives.asymmetric.rsa import RSAPrivateKey, RSAPublicKey
+from jwt import DecodeError
+
+__all__ = ["peek_claims", "decode", "encode", "authorization_header", "DecodeError"]
+
+
+def peek_claims(token: str) -> Mapping[str, str]:
+    """Returns the claims (payload) in the JWT token without validation.
+
+    These claims can be used to look up the correct key to use in :func:`decode`.
+    """
+    return pyjwt.decode(token, verify=False)
+
+
+def decode(
+    token: str,
+    key: bytes,
+    *,
+    audience: Union[None, str, bool] = None,
+    algorithms: Optional[List[str]] = None,
+) -> Mapping[str, str]:
+    """Returns the claims (payload) in the JWT token.
+
+    This will raise an exception if the claims can not be validated with the provided key.
+
+    :param audience: Set this to the audience you expect to be present in the claims.  Set
+       this to ``False`` to disable verifying the audience.
+    """
+    options = dict()
+    kwargs = dict()
+    if audience is False:
+        options["verify_aud"] = False
+    elif audience is True:
+        raise ValueError("audience can not be True")
+    elif audience is not None:
+        kwargs["audience"] = audience
+    return pyjwt.decode(token, key, options=options, algorithms=algorithms, **kwargs)
```

Comment:
mypy seems unhappy about the type of `**kwargs`.  I'm not sure what is up with it or what to do about that, any suggestions?

Reply:
The method signature for `decode` is:
```
def decode(self, jwt, key='', verify=True, algorithms=None, options=None, **kwargs):
```
and the mypy error is:
```
src/sentry/utils/jwt.py:47: error: Argument 5 to "decode" has incompatible type "**Dict[str, Union[str, bool]]"; expected "bool"
```
You can satisfy the checker by adding `verify=True` to the `decode` call:
```
return pyjwt.decode(token, key, verify=True, options=options, algorithms=algorithms, **kwargs)
```

Reply:
Thanks!  make sense now.